src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+
+      return builder.build();
+    }
+
+    case PredicateCondition::GreaterThan:
+      return sliced(PredicateCondition::GreaterThanEquals, _domain.next_value(*value));
+
+    case PredicateCondition::GreaterThanEquals: {
+      auto first_new_bin_id = _bin_for_value(*value);
+
+      if (first_new_bin_id == INVALID_BIN_ID) {
+        first_new_bin_id = _next_bin_for_value(*value);
+      }
+
+      DebugAssert(first_new_bin_id < bin_count(), "This should have been caught by _does_not_contain().");
+
+      GenericHistogramBuilder<T> builder{bin_count() - first_new_bin_id, _domain};
+
+      builder.add_sliced_bin(*this, first_new_bin_id, std::max(*value, bin_minimum(first_new_bin_id)),
+                             bin_maximum(first_new_bin_id));
+      builder.add_copied_bins(*this, first_new_bin_id + 1, bin_count());
+
+      return builder.build();
+    }
+
+    case PredicateCondition::BetweenInclusive:
+    case PredicateCondition::BetweenLowerExclusive:
+    case PredicateCondition::BetweenUpperExclusive:
+    case PredicateCondition::BetweenExclusive:
+      Assert(variant_value2, "BETWEEN needs a second value.");
+      return sliced(PredicateCondition::GreaterThanEquals, variant_value)
+          ->sliced(PredicateCondition::LessThanEquals, *variant_value2);
+
+    case PredicateCondition::Like:
+    case PredicateCondition::NotLike:
+      // TODO(anybody) Slicing for (NOT) LIKE not supported, yet
+      return clone();
+
+    case PredicateCondition::In:
+    case PredicateCondition::NotIn:
+    case PredicateCondition::IsNull:
+    case PredicateCondition::IsNotNull:
+      Fail("PredicateCondition not supported by Histograms");
+  }
+
+  Fail("Unreachable, but GCC does not realize...");
+}
+
+template <typename T>
+std::shared_ptr<AbstractStatisticsObject> AbstractHistogram<T>::scaled(const Selectivity selectivity) const {
+  GenericHistogramBuilder<T> builder(bin_count(), _domain);
+
+  // Scale the number of values in the bin with the given selectivity.
+  for (auto bin_id = BinID{0}; bin_id < bin_count(); bin_id++) {
+    builder.add_bin(bin_minimum(bin_id), bin_maximum(bin_id), bin_height(bin_id) * selectivity,
+                    _scale_distinct_count(selectivity, bin_height(bin_id), bin_distinct_count(bin_id)));
+  }
+
+  return builder.build();
+}
+
+template <typename T>
+std::shared_ptr<AbstractHistogram<T>> AbstractHistogram<T>::split_at_bin_bounds(
+    const std::vector<std::pair<T, T>>& additional_bin_edges) const {
+  if constexpr (std::is_same_v<
+                    T,
+                    pmr_string>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+    Fail("Cannot split_at_bin_bounds() on string histogram");
+  }
+
+  /**
+   * Create vector with pairs for each split.
+   * For a lower bin edge e, the new histogram will have to be split at previous_value(e) and e.
+   * For an upper bin edge e, the new histogram will have to be split at e and next_value(e).
```

Comment:
I didn't understand this part

Reply:
We can also discuss this in person and see if anything about this comment can be improved.

Reply:
First measure, I extended the doc of the function:
```c++
  /**
   * Derive a Histogram from this histograms splitting its bins so that both the current bounds as well as the bounds
   * specified in @param additional_bin_edges are present.
   * 
   * E.g., for a histogram with bins {[0, 10], [15, 20]}, split_at_bin_bounds({{-4, 5}, {16, 18}}) returns 
   * a histogram with bins {[0, 5], [6, 10], [15, 15], [16, 18], [19, 20]}
   * 
   * @param additional_bin_edges    Pair of minima and maxima of new bins
   */
  std::shared_ptr<AbstractHistogram<T>> split_at_bin_bounds(
      const std::vector<std::pair<T, T>>& additional_bin_edges) const;

```

Reply:
Also, I heavily rewrote the comments in the function and added examples. 

It is kind of hacky, I admit it, but Tim assured me it was pretty hard to cover all edge cases, so I am glad we have a solution that works at all.